internal/pkg/service/objectstorage/cloud/azure/storage.go

```diff
@@ -209,18 +210,51 @@ func (s AzureObjectStorageAdapter) CreateBucket(ctx context.Context, bucket *v1a
 // We want to prevent the Storage Account from being used by anyone
 func (s AzureObjectStorageAdapter) DeleteBucket(ctx context.Context, bucket *v1alpha1.Bucket) error {
 	storageAccountName := s.getStorageAccountName(bucket.Spec.Name)
-	// We delete the Storage Account, which delete the Storage Container
-	_, err := s.storageAccountClient.Delete(
+	// Check if Storage Account exists on Azure
+	existsStorageAccount, err := s.existsStorageAccount(ctx, storageAccountName)
+	if err != nil {
+		return err
+	}
+	// If Storage Account does not exists, we need to create it first
```

Comment:
This comment is confusing, we are in the delete method here why are we mentioning that the resource should be created first ?

Reply:
I think it wrongly copies something